Unable to build ROM with `GEN_LATEST` set to `3`

### Description

**System information**
- `make --version`: GNU Make 4.4.1
- `gcc --version`: arm-none-eabi-gcc (Arch Repository) 14.2.0
- `ld --version:` GNU ld (GNU Binutils) 2.43

**Make output**
```
/usr/lib/gcc/arm-none-eabi/14.2.0/cc1 -quiet <flags> -o build/modern/src/overworld.o src/overworld.c
src/overworld.c: In function 'UpdateTimeOfDay':
src/overworld.c:1577:72: error: division by zero [-Werror=div-by-zero]
 1577 |         gTimeBlend.weight = TIME_BLEND_WEIGHT(MORNING_HOUR_BEGIN, MORNING_HOUR_MIDDLE);
      |                                                                        ^
src/overworld.c:1585:90: error: division by zero [-Werror=div-by-zero]
 1585 |         gTimeBlend.weight = TIME_BLEND_WEIGHT(MORNING_HOUR_MIDDLE, MORNING_HOUR_END);
      |                                                                                          ^
src/overworld.c:1593:72: error: division by zero [-Werror=div-by-zero]
 1593 |         gTimeBlend.weight = TIME_BLEND_WEIGHT(EVENING_HOUR_BEGIN, EVENING_HOUR_END);
      |                                                                        ^
cc1: all warnings being treated as errors
make: *** [Makefile:394: build/modern/src/overworld.o] Error 1
make: *** Deleting file 'build/modern/src/overworld.o'
```

### Reproduction Steps

1. Change `GEN_LATEST` to `GEN_3` in [include/config/general.h](tree/master/include/config/general.h)
2. Run `make`

### Images / Video

_No response_

### Version

1.12.1 (Latest release)

### Upcoming/master Version

fa129ba906ea3a330f98d198eb3885f7c3fc24a7

### Discord contact info

_No response_

--- a/src/overworld.c
+++ b/src/overworld.c
@@ -916,7 +916,7 @@ static void LoadMapFromWarp(bool32 a1)
     ClearTempFieldEventData();
     ResetDexNavSearch();
     // reset hours override on every warp
-    sHoursOverride = 0; 
+    sHoursOverride = 0;
     ResetCyclingRoadChallengeData();
     RestartWildEncounterImmunitySteps();
 #if FREE_MATCH_CALL == FALSE
@@ -1559,7 +1559,7 @@ const struct BlendSettings gTimeOfDayBlend[] =
 };
 
 #define DEFAULT_WEIGHT 256
-#define TIME_BLEND_WEIGHT(begin, end) (DEFAULT_WEIGHT - (DEFAULT_WEIGHT * ((hours - begin) * MINUTES_PER_HOUR + minutes) / ((end - begin) * MINUTES_PER_HOUR)))
+#define TIME_BLEND_WEIGHT(begin, end) (DEFAULT_WEIGHT - (DEFAULT_WEIGHT * SAFE_DIV(((hours - begin) * MINUTES_PER_HOUR + minutes), ((end - begin) * MINUTES_PER_HOUR))))
 
 #define MORNING_HOUR_MIDDLE (MORNING_HOUR_BEGIN + ((MORNING_HOUR_END - MORNING_HOUR_BEGIN) / 2))
 

